### PR TITLE
Fixes #15360: Adding Windows shared folder feature to shared-file API

### PR DESCRIPTION
**Pull request by ArmandSy**
https://issues.rudder.io/issues/15360

### CODE REVIEW

**Comment by ArmandSy** [comment]
Commit modified

File: relay/sources/relayd/shared-folder/machineC/machineD/file [hunk 1]
```diff
@@ -0,0 +1 @@
+123
```
Comment (amousset) [inline relay/sources/relayd/shared-folder/machineC/machineD/file]: could you use actual machine ids that are in the test nodeslist.json?

File: relay/sources/relayd/src/shared_files.rs [hunk 2]
```diff
@@ -227,7 +221,8 @@ pub fn metadata_parser(buf: &mut warp::body::FullBody) -> Vec<String> {
             break;
         }
     }
-    metadata
+    println!("METADATA: {:?}", metadata);
```
Comment (amousset) [inline relay/sources/relayd/src/shared_files.rs]: this should be a proper log

Comment (amousset) [reply relay/sources/relayd/src/shared_files.rs]: probably at trace level

**Comment by ArmandSy** [comment]
Commit modified

**Comment by ArmandSy** [comment]
Commit modified

**Pull request merged**